samples/web/Continote/public/scripts/auth-helper.js

```diff
@@ -120,12 +222,32 @@ function AuthHelper(handleUserSignedIn, handleUserSignedOut) {
     // Configure FirebaseUI (if necessary) and then call the appropriate
     // callback for the current auth state.
     if (currentUserIsSignedIn_()) {
+      hideAllSignedOutOnlyElements_();
+
+      showAllSignedInOnlyElements_();
+
       handleUserSignedIn(currentUser_);
     } else {
+      hideAllSignedInOnlyElements_();
+
       // The current user is not signed in, so display a sign in UI using
       // FirebaseUI.
-      firebaseUi_.reset();
-      firebaseUi_.start("#" + firebaseUiSignInContainerId_, firebaseUiConfig_);
+      var signInUiContainer = document.getElementById(signInUiContainerId_);
+      if (signInUiContainer) {
+        // If the sign in container has no children it has not been populated
+        // with the sign in UI, so we add the sign in UI to it.
+        if (!signInUiContainer.firstElementChild) {
+          Utils.appendElementBasedOnTemplateToContainer(
+              signInUITemplate_, signInUiContainer);
+        }
+
+        // Now that the sign in UI is certainly on the page, set up FirebaseUI.
+        firebaseUi_.reset();
```

Comment:
Although "reset" isn't required, it is suggested within the FirebaseUI-Web README when re-rendering the widget. See: https://github.com/firebase/firebaseui-web#known-issues